src/kernel.ts

```diff
@@ -173,17 +171,17 @@ class Kernel {
     return utils.ajaxRequest(kernelServiceUrl, {
       method: "GET",
       dataType: "json"
-    }).then((success: utils.IAjaxSuccess): IKernelId[] => {
-      if (success.xhr.status === 200) {
-        if (!Array.isArray(success.data)) {
-          throw Error('Invalid kernel list');
-        }
-        for (var i = 0; i < success.data.length; i++) {
-          validateKernelId(success.data[i]);
-        }
-        return success.data;
+    }).then((success: utils.IAjaxSuccess) => {
+      if (success.xhr.status !== 200) {
+        throw Error('Invalid Status: ' + success.xhr.status);
+      }
+      if (!Array.isArray(success.data)) {
+        throw Error('Invalid kernel list');
+      }
+      for (var i = 0; i < success.data.length; i++) {
+        validate.validateKernelId(success.data[i]);
       }
-      throw Error('Invalid Status: ' + success.xhr.status);
+      return <IKernelId[]>success.data;
```

Comment:
I move this from the explicit return type to the cast because its more indicative of what is actually happening. By omitting the return type, the compiler will pick it up automatically. I applied this convention throughout this diff.